### DISCUSSION TITLE
[Medusa] 405 during admin login: following "Deploy Admin to Vercel" guide

### DISCUSSION THREAD

**Discussion by hash4di** [vercel/community#5296] [Help] [answered]
### Summary

Hello team!

At the beginning I would like to say "Thank you!" to everyone involved in this project, it's a great tool. I love the UI and the documentation.

I would really appreciate the support here.

***Issue:***

`POST https://medusa-vercel.vercel.app/admin/auth 405 (Method Not Allowed)`

I followed the [Deploy Admin to Vercel | Medusa instruction https://docs.medusajs.com/deployments/admin/deploying-on-vercel](https://docs.medusajs.com/deployments/admin/deploying-on-vercel)

After successful deployment I have working *Medusa* locally, both `start` and `develop` options. I am able to successfully login.

But I am not able to login to admin backend on app deployed to Vercel. Dunno why.
I have read all the topics in *Discussions* both Vercel and Medusa GH orgs.

Vercel Backend: [https://medusa-vercel.vercel.app/login](https://medusa-vercel.vercel.app/login)
Credentials, standard: admin@medusa-test.com
Pass: supersecret

![Screenshot 2023-12-26 at 14 34 48](https://github.com/vercel/community/assets/218911/afe36be6-ad47-492b-810d-b74bff0edc0f)

Repository, open: [https://github.com/hash4di/medusa-vercel](https://github.com/hash4di/medusa-vercel)

```
ADMIN_CORS=https://medusa-vercel.vercel.app
MEDUSA_ADMIN_BACKEND_URL=https://medusa-vercel.vercel.app
```

Any ideas?

Kind Regards,
h4

### Example

https://medusa-vercel.vercel.app/login

### Steps to Reproduce

Use credentials:
l: admin@medusa-test.com
p: supersecret

It's throwing `POST https://medusa-vercel.vercel.app/admin/auth 405 (Method Not Allowed)`.

**Comment by amyegan** [comment]
Hi @hash4di! The most common reason I've seen for this kind of error is a missing or invalid [environment variable](https://vercel.com/docs/projects/environment-variables). Please double check that all necessary environment variables have the correct names and values in project settings, redeploy, and let me know if that solves it for you. 🙏

**Comment by lekterable** [comment]
also experiencing this now after deploying the template nextjs + medusa project to vercel for the first time, all my env vars:

<img width="346" alt="image" src="https://github.com/vercel/community/assets/23037261/c0b97de7-4d5c-43fd-892e-9f94b70d7832">

I confirmed that they are correct and I'm not using Redis

**Comment by kelvinnmugendi** [comment]
@hash4di @lekterable Did you find a solution to this issue?

**Reply by lekterable** [reply, marked as answer]
@kevinrobert3 I did eventually, for me it was a misconception of how to deploy it properly.

You basically need 2 separate Vercel projects - 1 for the admin and 1 for the storefront, I'm sharing my build settings below hoping it will help someone:

1. Admin
<img width="894" alt="image" src="https://github.com/vercel/community/assets/23037261/3e5d4c17-0255-4c7d-a06f-17d5803e5c41">

2. Storefront
<img width="895" alt="image" src="https://github.com/vercel/community/assets/23037261/ab1e07fa-e596-4453-9b92-829e4f83d491">

And of course you also need a server deployed somewhere else e.g. Railway to handle all the requests + make sure all the variables like `ADMIN_CORS`, `NEXT_PUBLIC_MEDUSA_BACKEND_API` etc. point at the correct domains.

**Reply by thomcrielaard** [reply]
Could you elaborate a bit more? I'm currently facing this same issue, but I don't see why the frontend is important or could solve this issue (it didn't for me). The issue occurs on the backend admin panel from Vercel, when attempting to make a request to log in. I'm assuming this is solved by using "Railway to handle all requests", but what exactly do you mean by this? What should I deploy to Railway, and what is the point of deploying this to Vercel if I need an additional hosting service? 

Additionally you mentioned to make sure to point all the variables to the correct domains, where can I check if they are correct?

**Reply by lekterable** [reply]
@thomcrielaard 

> What should I deploy to Railway, and what is the point of deploying this to Vercel if I need an additional hosting service?

Vercel hosts only your static pages like the build folder from Next.js, you need Railway to deploy the Node.js server which handles all BE requests

Vercel: Storefront + Admin
Railway: Medusa Node.js server

> Where can I check if they are correct?

Well, that's kinda your job, for example `ADMIN_CORS` is your admin website e.g `admin.store.com` while `NEXT_PUBLIC_MEDUSA_BACKEND_API` could be something like `server.page.com` depends on how you setup your domains
